src/runtime/print.go

```diff
@@ -39,7 +39,7 @@ func printuint16(n uint16) {
 	printuint32(uint32(n))
 }
 
-func printint16(n uint16) {
+func printint16(n int16) {
```

Comment:
Turns out this function has been relying on a bug in the compile all this time...